d2/src/main/java/com/linkedin/d2/xds/XdsClientImpl.java

```diff
@@ -389,6 +412,13 @@ private void onReconnect()
         watcher.onReconnect();
       }
     }
+
+    private void onRemoval(){
+      for (ResourceWatcher watcher : _watchers)
```

Comment:
let's add detailed comment here about why we choose to notify the watchers with the last known data instead of setting the _data to null.

Reply:
Thanks. Add details here